app/src/androidTest/java/com/github/sdp_begreen/begreen/viewModels/ContestCreationViewModelGeoTest.kt

```diff
@@ -0,0 +1,124 @@
+package com.github.sdp_begreen.begreen.viewModels
+
+import android.location.Address
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.SmallTest
+import com.github.sdp_begreen.begreen.models.CustomLatLng
+import com.github.sdp_begreen.begreen.rules.KoinTestRule
+import com.github.sdp_begreen.begreen.services.GeocodingService
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.After
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+class ContestCreationViewModelGeoTest {
+
+    companion object {
+
+        private var geo: GeocodingService = mock()
+        private lateinit var vm: ContestCreationViewModel
+
+        private val address = Address(null).apply {
+            adminArea = "Vaud"
+            countryCode = "CH"
+            locality = "Lausanne"
+            postalCode = "1010"
+            latitude = 46.519653
+            longitude = 6.632273
+        }
+
+        private val customLatLng = CustomLatLng(46.519653, 6.632273)
+
+        private val listOfAddress = listOf(address)
+
+        @BeforeClass
+        @JvmStatic
+        fun setUpGeo() {
+            runTest {
+                whenever(geo.getAddresses(any(), any()))
+                    .thenReturn(listOfAddress)
+
+                whenever(geo.getLongLat(any()))
+                    .thenReturn(customLatLng)
+
+            }
+        }
+    }
+
+
+    @get:Rule
+    val koinTestRule = KoinTestRule(
+        modules = listOf(module {
+            single { geo }
+        })
+    )
+
+    @Before
+    fun setUp() {
+        vm = ContestCreationViewModel()
+    }
+
+    @After
+    fun tearDown() {
+        geo = mock()
+        setUpGeo()
+    }
+
+    //@Test
```

Comment:
Why is this test commented ? The next one `isEditCountryCorrect` seems to work correctly and is similar. If you forgot to put it back maybe you could do it. Otherwise, if it's failing for no reason, you should mention it with comments